flotilla-server/main.go

```diff
@@ -25,12 +28,31 @@ func main() {
 	}
 
 	d, err := daemon.NewDaemon(config)
+
 	if err != nil {
 		panic(err)
 	}
 
+	// Connect to coordinator service to establish our presence now that we
+	// are up and running ok.
+	// TODO: Modify this so that we can do TLS connections if we have the right
+	// information
+	// TODO: Consider go func () ... the actual registration
+	var client coordinate.Client
+	if coordinator != nil && flota != nil {
```

Comment:
Are these pointers *ever* `nil`? I suspect you actually want to check the values:

```go
if *coordinator != "" && *flota != "" {
...
}
```